src/codecs/framers/newline_delimited.rs

```diff
@@ -0,0 +1,140 @@
+use crate::codecs::{BoxedFramer, BoxedFramingError, CharacterDelimitedCodec, FramingConfig};
+use bytes::{Bytes, BytesMut};
+use serde::{Deserialize, Serialize};
+use tokio_util::codec::Decoder;
+
+/// Config used to build a `NewlineDelimitedCodec`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct NewlineDelimitedDecoderConfig {
+    /// The maximum length of the byte buffer.
+    ///
+    /// This length does *not* include the trailing delimiter.
+    max_length: Option<usize>,
+}
+
+impl NewlineDelimitedDecoderConfig {
+    /// Creates a new `NewlineDelimitedDecoderConfig`.
+    pub const fn new() -> Self {
+        Self { max_length: None }
+    }
+}
+
+#[typetag::serde(name = "newline_delimited")]
+impl FramingConfig for NewlineDelimitedDecoderConfig {
+    fn build(&self) -> crate::Result<BoxedFramer> {
+        Ok(Box::new(
+            self.max_length
+                .map(NewlineDelimitedCodec::new_with_max_length)
+                .unwrap_or_else(NewlineDelimitedCodec::new),
+        ))
+    }
+}
+
+/// A codec for handling bytes that are delimited by (a) newline(s).
+#[derive(Debug, Clone)]
+pub struct NewlineDelimitedCodec(CharacterDelimitedCodec);
+
+impl NewlineDelimitedCodec {
+    /// Creates a new `NewlineDelimitedCodec`.
+    pub const fn new() -> Self {
+        Self(CharacterDelimitedCodec::new('\n'))
+    }
+
+    /// Creates a `NewlineDelimitedCodec` with a maximum frame length limit.
+    ///
+    /// When more bytes than `max_length` have been read, all bytes will be
+    /// discarded until reaching the next newline.
```

Comment:
This may be my poor grasp of the english language but... this is saying:

```
(read all these bytes)[limit!]<discard until...>\n
(read all these bytes)[limit!]<discard until...>\n
```

_Not_:

```
(read all these bytes)[limit!]<discard>\n
<discard>\n
```

Reply:
I rephrased this a couple of time because I was having a hard time to succinctly express it, apparently still not good enough 😄

Do you have a suggestion?

Reply:
Not really 😆 I think it's accurate, but I'd not call my _technical_ english very strong.

Reply:
Maybe something like the following?

> Any frames longer than `max_length` bytes will be discarded entirely.

Basically, with `max_length = 3`, we should see behavior like:

```
xxx\n   <- kept
xxxx\n  <- dropped
xx\n    <- kept
```

Reply:
Short and clear 👍😎

Reply:
Changed in b2220aa94048b06450af996eb8222aa675947d07